src/pages/HomePage.tsx

```diff
@@ -35,20 +50,24 @@ const HomePage = ({ streamersById, featuredStreamers, onData }: { videosByDate:
 const mapStateToProps = (state: { mainReducer: State }) => {
     return {
         streamersById: state.mainReducer.streamersById,
-        featuredStreamers: state.mainReducer.featuredStreamers
+        featuredStreamers: state.mainReducer.featuredStreamers,
+        latestVideosById: state.mainReducer.latestVideosById,
+        latestVideos: state.mainReducer.latestVideos
     }
 }
 
 const mapDispatchToProps = (dispatch: (arg0: any) => {}) => {
     return {
-        onData: (data: any) => {
+        onDataFeatured: (data: any) => {
             dispatch(addStreamersById(data))
+        },
+        onDataLatest: (data: any) => {
+            dispatch(addLatestVideosById(data))
```

Comment:
probably should be without ById as it has both. streamers above currently relying on hardcoded stuff which i ll need to remove and create diff method or rename

Reply:
u are missing other part u are missing split to diff parts check streamer videos it has the way to split. im not sure if its efficient and may be in the future ill merge everything into one action and do all the splitting in the reducer